docs/guides/server/caching.adoc

```diff
@@ -280,6 +280,39 @@ The keys and the certificate are stored in the database so they are available to
 By default, the certificate is valid for 60 days and is rotated at runtime every 30 days.
 Use the option `cache-embedded-mtls-rotation-interval-days` to change this.
 
+=== Running inside a service mesh
+
+When using a service mesh like Istio, you might need to allow a direct mTLS communication between the {project_name} Pods to allow for the mutual authentication to work.
+Otherwise, you might see error messages like `JGRP000006: failed accepting connection from peer SSLSocket` that indicate that a wrong certificate was presented, and the cluster will not form correctly.
+
+You then have the option to allow direct mTLS communication between the {project_name} nodes, or rely on the service mesh transport security to encrypt the communication and to authenticate peers.
+
+To allow direct mTLS communication for {project_name} when using Istio:
+
+* Apply the following configuration to allow direct communication.
++
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: infinispan-allow-nomtls
+spec:
+  selector:
+    matchLabels:
+      app: keycloak # <1>
+  portLevelMtls:
+    "7800": # <2>
+      mode: PERMISSIVE
+----
+<.> Update the labels to match your {project_name} deployment.
+<.> Port 7800 is the default. Adjust it if you change the data transmission port.
+
+As an alternative, to disable the mTLS communication, and rely on the service mesh to encrypt the traffic:
+
+* Set the option `cache-embedded-mtls-enabled` to `false`.
+* Configure your service mesh to authorize only traffic from other {project_name} nodes for the data transmission port (default: 7800).
```

Comment:
```suggestion
* Configure your service mesh to authorize only traffic from other {project_name} Pods for the data transmission port (default: 7800).
```

Reply:
I tried to describe this in dependent of Istio for a general service mesh, so I avoided "Pod" here and used "node". Let me hear your thoughts.

Reply:
Isn't the `PeerAuthentication` CR specific to Kubernetes? Or is it possible to use the same configuration also for other types of deployment?

Reply:
The whole section looks very "Kubernetes specific" by mentioning Pods and CRs.

Reply:
Yes, `PeerAuthentication` is specific to Kubernetes as well as Istio is AFAIK an Kubernetes only thing. Still the sentence above tries to give a general advice: 

> You then have the option to allow direct mTLS communication between the {project_name} nodes, or rely on the service mesh transport security to encrypt the communication and to authenticate peers.

And after that is then the Istio specific guidance that is "community confirmed" to work. Not ideal, still can't think of a better way unless we split things up which comes at a different cost. Hm.

Reply:
So would it be safe to assume that a service mesh is always a Kubernetes thing? I suppose you got me convinced here...

Reply:
I would vote for changing all nodes to Pods. Kubernetes is the deployment we know is failing and we want to address the reported issue. If another issue comes up we can generalize later.